lib/ansible/module_utils/network/vyos/config/lldp_interfaces/lldp_interfaces.py

```diff
@@ -0,0 +1,386 @@
+            result['changed'] = True
+
+        result['commands'] = commands
+
+        if self._module._diff:
+            result['diff'] = resp['diff'] if result['changed'] else None
+
+        changed_lldp_interfaces_facts = self.get_lldp_interfaces_facts()
+        result['before'] = existing_lldp_interfaces_facts
+        if result['changed']:
+            result['after'] = changed_lldp_interfaces_facts
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_lldp_interfaces_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = self._module.params['config']
+        have = existing_lldp_interfaces_facts
+        resp = self.set_state(want, have)
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        state = self._module.params['state']
+        if state == 'overridden':
+            commands.extend(self._state_overridden(want, have))
+        elif state == 'deleted':
+            if want:
+                for item in want:
+                    name = item['name']
+                    obj_in_have = search_obj_in_list(name, have)
+                    commands.extend(self._state_deleted(obj_in_have))
+            else:
+                for have_item in have:
+                    commands.extend(self._state_deleted(have_item))
+        elif state == 'merged':
+            for want_item in want:
+                name = want_item['name']
+                have_item = search_obj_in_list(name, have)
+                commands.extend(self._state_merged(want_item, have_item))
+        elif state == 'replaced':
+            for want_item in want:
+                name = want_item['name']
+                obj_in_have = search_obj_in_list(name, have)
+                commands.extend(self._state_replaced(want_item, obj_in_have))
+        return commands
+
+    @staticmethod
+    def _state_replaced(want, have):
+        """ The command generator when state is replaced
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        if have:
+            commands.extend(Lldp_interfaces._render_del_commands(want, have))
+        commands.extend(Lldp_interfaces._state_merged(want, have))
```

Comment:
The method can be converted to a generic class method `_state_replaced(self, want, have)`. Remove `@staticmethod`. Applicable at other places as well.

Reply:
Done